src/libs/PhoneNumber.ts

```diff
@@ -0,0 +1,42 @@
+// eslint-disable-next-line no-restricted-imports
+import {parsePhoneNumber as originalParsePhoneNumber, ParsedPhoneNumber, ParsedPhoneNumberInvalid, PhoneNumberParseOptions} from 'awesome-phonenumber';
+import CONST from '@src/CONST';
+
+/**
+ * Wraps awesome-phonenumber's parsePhoneNumber function to handle the case where we want to treat
+ * a US phone number that's technically valid as invalid. eg: +115005550009.
+ * See https://github.com/Expensify/App/issues/28492
+ */
+function parsePhoneNumber(phoneNumber: string, options?: PhoneNumberParseOptions): ParsedPhoneNumber {
+    const parsedPhoneNumber = originalParsePhoneNumber(phoneNumber, options);
+    if (!parsedPhoneNumber.possible) {
+        return parsedPhoneNumber;
+    }
+
+    const phoneNumberWithoutSpecialChars = phoneNumber.replace(CONST.REGEX.SPECIAL_CHARS_WITHOUT_NEWLINE, '');
+    if (!/^\+11[0-9]{10}$/.test(phoneNumberWithoutSpecialChars)) {
+        return parsedPhoneNumber;
+    }
+
+    const countryCode = phoneNumberWithoutSpecialChars.substring(0, 2);
+    const phoneNumberWithoutCountryCode = phoneNumberWithoutSpecialChars.substring(2);
+
+    return {
+        ...parsedPhoneNumber,
+        valid: false,
+        possible: false,
+        number: {
+            ...parsedPhoneNumber.number,
+
+            // mimic the behavior of awesome-phonenumber
+            e164: phoneNumberWithoutSpecialChars,
+            international: `${countryCode} ${phoneNumberWithoutCountryCode}`,
+            national: phoneNumberWithoutCountryCode,
+            rfc3966: `tel:${countryCode}-${phoneNumberWithoutCountryCode}`,
+            significant: phoneNumberWithoutCountryCode,
+        },
+    } as ParsedPhoneNumberInvalid;
+}
+
+// eslint-disable-next-line import/prefer-default-export
+export {parsePhoneNumber};
```

Comment:
I wasn't very happy with the `export {*} from 'awesome-phone-number'` part. I I think it was unnecessary and added complexity to it. I'm now only exporting `parsePhoneNumber` and I updated the lint rule to only restrict this function from `awesome-phone-number` (we are not using anything else throughout the entire codebase anyways).